File: app/lib/models/task.dart
class Task {
  final String id;
  final String title;
  final String description;
  final String priority;
  bool isCompleted;
  final String? dueDate;
  final String? dueTime;
  final bool isPinned;
  final String? listId;
  final List<TagInfo> tags;
  final ListInfo? list;
  final SubTaskCount? subTaskCount;

  Task({
    required this.id,
    required this.title,
    this.description = '',
    this.priority = 'none',
    this.isCompleted = false,
    this.dueDate,
    this.dueTime,
    this.isPinned = false,
    this.listId,
    this.tags = const [],
    this.list,
    this.subTaskCount,
  });

  factory Task.fromJson(Map<String, dynamic> json) => Task(
    id: json['id'],
    title: json['title'],
    description: json['description'] ?? '',
    priority: json['priority'] ?? 'none',
    isCompleted: json['isCompleted'] ?? false,
    dueDate: json['dueDate'],
    dueTime: json['dueTime'],
    isPinned: json['isPinned'] ?? false,
    listId: json['listId'],
    tags: (json['tags'] as List<dynamic>?)?.map((t) => TagInfo.fromJson(t)).toList() ?? [],
    list: json['list'] != null ? ListInfo.fromJson(json['list']) : null,
    subTaskCount: json['subTaskCount'] != null ? SubTaskCount.fromJson(json['subTaskCount']) : null,
  );
}

class TagInfo {
  final String id;
  final String name;
  final String color;
  TagInfo({required this.id, required this.name, required this.color});
  factory TagInfo.fromJson(Map<String, dynamic> json) => TagInfo(
    id: json['id'], name: json['name'], color: json['color'] ?? '#a855f7',
  );
}

class ListInfo {
  final String id;
  final String name;
  final String color;
  final String icon;
  ListInfo({required this.id, required this.name, required this.color, required this.icon});
  factory ListInfo.fromJson(Map<String, dynamic> json) => ListInfo(
    id: json['id'], name: json['name'], color: json['color'] ?? '#6366f1', icon: json['icon'] ?? 'list',
  );
}

class SubTaskCount {
  final int total;
  final int completed;
  SubTaskCount({required this.total, required this.completed});
  factory SubTaskCount.fromJson(Map<String, dynamic> json) => SubTaskCount(
    total: json['total'] ?? 0, completed: json['completed'] ?? 0,
  );
}

File: web/shared/types/task.ts
import type { Priority } from './enums.js';

export interface Task {
  id: string;
  userId: string;
  listId: string | null;
  title: string;
  description: string;
  priority: Priority;
  isCompleted: boolean;
  completedAt: string | null;
  dueDate: string | null;
  dueTime: string | null;
  sortOrder: number;
  isPinned: boolean;
  deletedAt: string | null;
  createdAt: string;
  updatedAt: string;
  /** 关联数据 —— 仅在详情接口返回 */
  tags?: TagInfo[];
  subTasks?: SubTaskInfo[];
  subTaskCount?: { total: number; completed: number } | null;
  reminders?: ReminderInfo[];
  list?: ListInfo;
}

export interface CreateTaskDto {
  title: string;
  description?: string;
  listId?: string | null;
  priority?: Priority;
  dueDate?: string | null;
  dueTime?: string | null;
  tagIds?: string[];
}

export interface UpdateTaskDto {
  title?: string;
  description?: string;
  listId?: string | null;
  priority?: Priority;
  isCompleted?: boolean;
  dueDate?: string | null;
  dueTime?: string | null;
  sortOrder?: number;
  isPinned?: boolean;
}

export interface TaskFilterDto {
  listId?: string;
  priority?: Priority;
  status?: 'all' | 'active' | 'completed';
  keyword?: string;
  dateFrom?: string;
  dateTo?: string;
  tagIds?: string[];
  page?: number;
  pageSize?: number;
  sortBy?: 'createdAt' | 'updatedAt' | 'dueDate' | 'priority' | 'sortOrder';
  sortOrder?: 'asc' | 'desc';
  isPinned?: boolean;
}

export interface BatchTaskDto {
  action: 'delete' | 'complete' | 'updatePriority' | 'moveToList' | 'updateTags';
  taskIds: string[];
  payload?: {
    priority?: Priority;
    listId?: string;
    tagIds?: string[];
  };
}

export interface MoveTaskDto {
  targetListId: string | null;
}

export interface BindTagsDto {
  tagIds: string[];
}

// ---- 轻量关联类型（详情中嵌套用） ----

export interface TagInfo {
  id: string;
  name: string;
  color: string;
}

export interface SubTaskInfo {
  id: string;
  text: string;
  isCompleted: boolean;
  sortOrder: number;
}

export interface ReminderInfo {
  id: string;
  remindAt: string;
  isRecurring: boolean;
  recurType: string;
  recurRule: string | null;
  isTriggered: boolean;
}

export interface ListInfo {
  id: string;
  name: string;
  color: string;
  icon: string;
}
